Deploy Sprint 2 W2

nicoloboo02 opened the pull request:
# Description
This pull request includes all the features that have been implemented during the first week of Sprint 2

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Screenshot(If applies):

Manuelgruiz committed:
feat:Add Modal  #19

Manuelgruiz committed:
feat:Fix modal layers #19

Manuelgruiz committed:
Fixingo:Fix modal text  #19

Manuelgruiz committed:
Merge branch 'develop' into feature/19-Intervention_creation_view

Manuelgruiz committed:
feat:Add base modal#19

nicoloboo02 committed:
feat: Export beneficiaries data #33

A button has been added to export
 the data of the beneficiaries to excel.

nicoloboo02 committed:
feat: Export interventions #33

A button has been added to export
the intervention data to excel.

Ronald752 committed:
feat: Add beneficiary details

sebasruii committed:
refactor: Create user using API #36

sebasruii committed:
refactor: Login using API #36

Manuelgruiz committed:
feat:Add sidebar link #19

Manuelgruiz committed:
fix:Fix Modal Problems  #19

sebasruii committed:
Merge branch 'main' of github.com:ISPP-07/frontend-acat into feature/36-login_backend_api

nicoloboo02 committed:
feat: Add export interventions #33

nicoloboo02 committed:
feat: Add import beneficiaries #33

sebasruii committed:
refactor: Add redirect to main page #36

sebasruii committed:
refactor: Show messages #36

sebasruii committed:
refactor: Create beneficiary using API #36

Ronald752 committed:
feat: Add attribute name changes

nicoloboo02 committed:
fix: Remove interventions column #33

sebasruii committed:
refactor: Redirect to beneficiaries #36

sebasruii committed:
refactor: Add password validation #36

Manuelgruiz committed:
feat:Add beneficiaries connection #38

nicoloboo02 committed:
test: Solve components tests #36

nicoloboo02 committed:
Merge pull request #39 from ISPP-07/feature/36-login_backend_api

Feature/36 login backend api

nicoloboo02 committed:
Merge branch 'develop' into feature/33-Export_data_to_excel

sebasruii committed:
test: Add tests for #33

sebasruii committed:
Merge pull request #35 from ISPP-07/feature/33-Export_data_to_excel

Feature/33 export data to excel

Ronald752 committed:
feat: Add error handling

JMGarCas committed:
feat: sidebar is now responsive

JMGarCas committed:
feat: CreateUserForm is now responsive

JMGarCas committed:
feat: Login form has been refactored

JaviGarc1a committed:
feat: Buttons component and CardBeneficiary

JaviGarc1a committed:
feat: CardUser

JaviGarc1a committed:
feat: Serachbar component

JaviGarc1a committed:
feat: CardIntervention component

JaviGarc1a committed:
Merge pull request #43 from ISPP-07/feature/30-refactoring_and_responsiveness_12

Feature/30 refactoring and responsiveness 12

JaviGarc1a committed:
Merge pull request #42 from ISPP-07/feature/31-refactoring_and_responsiveness_22

Feature/31 refactoring and responsiveness (2/2)

JaviGarc1a committed:
feat: Add new files and delete unused files

JMGarCas committed:
feat: create modal has been refactored and is responsive

JaviGarc1a committed:
Merge branch 'feature/41-join_new_components' into feature/30-refactoring_and_responsiveness_12

JaviGarc1a committed:
Merge pull request #44 from ISPP-07/feature/30-refactoring_and_responsiveness_12

feat: create modal has been refactored and is responsive

Manuelgruiz committed:
fix:Solve vercel error #19

nicoloboo02 committed:
Merge branch 'develop' into feature/19-Intervention_creation_view

Manuelgruiz committed:
fix:Solve vercel erros #38

Ronald752 committed:
feat: Add alert of error

Ronald752 committed:
feat: Add alert

JaviGarc1a committed:
feat: Update clock.svg and cardIntervention

Manuelgruiz committed:
fix:Solve vercel problems  #19

JMGarCas committed:
feat: beneficiary detailed refactored responsive

JMGarCas committed:
Merge branch 'feature/41-join_new_components' of https://github.com/ISPP-07/frontend-acat into feature/41-join_new_components

JMGarCas committed:
feat: parse date beneficiary detail

sebasruii committed:
Merge pull request #21 from ISPP-07/feature/19-Intervention_creation_view

Feature/19 register intervention

JMGarCas committed:
feat: LoginPage responsive

JaviGarc1a committed:
feat: Update calendar.svg and cardIntervention.jsx components

JMGarCas committed:
feat: toggle modal beneficiaries

JaviGarc1a committed:
refactor: Remove unused code and update text in Searchbar component

JaviGarc1a committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/41-join_new_components' into feature/41-join_new_components

JaviGarc1a committed:
fix: Fix deploy

JaviGarc1a committed:
fix: URL 404 error

JaviGarc1a committed:
feat: Fetfch intervention

JMGarCas committed:
feat: Intervention details responsive refactor

JaviGarc1a committed:
Merge remote-tracking branch 'origin/develop' into feature/41-join_new_components

JaviGarc1a committed:
feat: Update import paths and error handling in beneficiaries and interventions pages

JaviGarc1a committed:
fix: Add imports

JaviGarc1a committed:
fix: reset modal

JaviGarc1a committed:
feat: Add plus icon

JMGarCas committed:
feat: form create beneficiary responsive

JaviGarc1a committed:
fix: Changes birth-date to birth_date

sebasruii committed:
Merge pull request #45 from ISPP-07/feature/41-join_new_components

Responsiveness app

Ronald752 committed:
Merge branch 'develop' into feature/34-Connect_to_backend

Manuelgruiz committed:
fix:Solve vercel problems #38

Manuelgruiz committed:
fix:Solve vercel problems  #38

Ronald752 committed:
Merge branch 'develop' into feature/38-Connect_list_entities

Ronald752 committed:
feat: Add list beneficiaries and interventions

Ronald752 committed:
fix: Solve listing error

Ronald752 committed:
fix: Solve detail view error

Ronald752 committed:
feat: Add change mockapi

sebasruii committed:
Merge pull request #37 from ISPP-07/feature/34-Connect_to_backend

Feature/34 connect to backend

Ronald752 committed:
Merge branch 'develop' into feature/38-Connect_list_entities

sebasruii committed:
Merge pull request #40 from ISPP-07/feature/38-Connect_list_entities

Feature/38 connect entities  to backend

nicoloboo02 committed:
test: Upgrade tests #48

JMGarCas committed:
fix: Searchbar button now works in mobile

Ronald752 committed:
Merge pull request #49 from ISPP-07/feature/48-Test_refactor

test: Upgrade tests #48

Ronald752 committed:
feat: Add connection of interventions to the backend

Ronald752 committed:
Merge pull request #58 from ISPP-07/fix/55-Searchbar_modal_buttons

Fix/50 searchbar modal button

Ronald752 committed:
Merge branch 'develop' into fix/59-Fix_create_intervention

sebasruii committed:
feat: Select beneficiary #59

sebasruii committed:
Merge pull request #60 from ISPP-07/fix/59-Fix_create_intervention

feat: Add connection of interventions to the backend

JMGarCas committed:
feat: initial blueplrints (lacks endpoints)

JMGarCas committed:
feat: date by default

JaviGarc1a committed:
feat: Add responsive behavior to login form

JaviGarc1a committed:
fix: Delete sidebar entries to modal

JaviGarc1a committed:
feat: Add function to hide/show sidebar depending screen layout

JMGarCas committed:
feat: tipology dropdown

JaviGarc1a committed:
fix: Add validation window is exist

JMGarCas committed:
style: no form when edit view is not toggled

If you change some attributes in "edit mode" and then you come back to "view mode", the changes you have made seem like they have been applied, which could confuse the users.

JMGarCas committed:
Merge pull request #63 from ISPP-07/fix/56-fix_sidebar_links_to_creation

Fix/56 fix sidebar links to creation

Ronald752 committed:
fix: Change of styles

JMGarCas committed:
style: refactor Select so its similar to other components

JMGarCas committed:
fix: suggested changes

JMGarCas committed:
Merge pull request #64 from ISPP-07/fix/62-Fix_intervention_form_style

fix: Change of styles

JaviGarc1a committed:
fix: Backend change URL

JaviGarc1a committed:
styles: Conditional rendering to edit mode in patient

JaviGarc1a committed:
feat: Add prop isSubmit to ButtonIcon

JaviGarc1a committed:
feat: Add confirmation modal

JaviGarc1a committed:
feat: Edit and delete patient

JaviGarc1a committed:
fix: Responsive modal confirmation

JaviGarc1a committed:
fix: Add toggle delete in button icon

JaviGarc1a committed:
fix: Load page when edit

JaviGarc1a committed:
feat: Toggle button to rehabilitate patient

JaviGarc1a committed:
fix: Fix style toggle button

JaviGarc1a committed:
fix: Add validation DNI, NIE and passport

JaviGarc1a committed:
fix: form validation and error handling

JaviGarc1a committed:
feat: Update beneficiary birth date and first technician

JaviGarc1a committed:
fix: Sidebar show or hidden depend screen layout

JaviGarc1a commented on src/app/components/CreateUserForm.jsx:
```diff
@@ -2,29 +2,79 @@
+
+			axios
+				.post(
+					process.env.NEXT_PUBLIC_BASE_URL + '/shared/user/',
```
Create user has wrong URL, so can't post. New URL is /shared/auth/login

nicoloboo02 replied:
Buenas Javi, esto no es así. La url que tu me indicas es para loguearse, con lo que esto no sería un error.

nicoloboo02 replied:
Good afternoon Javi, this is not the case. The url that you indicate is for logging in, so this would not be an error.

JaviGarc1a commented on src/app/beneficiaries/create.jsx:
```diff
@@ -1,192 +1,441 @@
+						<div className="relative flex items-center border-2 rounded-xl border-gray-200 bg-white">
+							<input
+								className="flex items-center rounded-xl p-1 w-full"
+								type="date"
```
To create an intervention, it must have an input type datetime and no input type date, since both are registered.

JaviGarc1a commented on src/app/beneficiaries/create.jsx:
```diff
@@ -1,192 +1,441 @@
+			})
+			.catch(function (error) {
+				alert(
+					`Ha habido un error al crear al nuevo beneficiario: ${error.response.data.detail}`
```
Creating a beneficiary with invalid data will result in this error. It's not very descriptive:
![image](https://github.com/ISPP-07/frontend-acat/assets/72880234/39a0ef4c-36d4-47fe-a2d5-1d2bf3d697ad)

JaviGarc1a commented on src/app/components/LoginForm.jsx:
```diff
@@ -3,29 +3,53 @@
+				router.push(`/beneficiaries?showSidebar=${stateSidebar}`)
+			})
+			.catch(function (error) {
+				alert('Error al iniciar sesión: ' + error.response.data.detail)
```
Error if you log in with a user that does not exist:
![image](https://github.com/ISPP-07/frontend-acat/assets/72880234/8a5cac9d-0332-4f9e-bc42-06fff0246b09)

JaviGarc1a reviewed (commented):
When you click on the Beneficiary card, the detail view will be white. This bug has been fixed in another feature, but it's not merge.

Manuelgruiz committed:
feat:Add rehabilited screen #53

sebasruii committed:
feat: Add (C)RUD operations for users #51

sebasruii commented on src/app/interventions/page.jsx:
![image](https://github.com/ISPP-07/frontend-acat/assets/73228845/d351c293-cb41-45b4-88ee-d670679a7572)
Import button style should be corrected

sebasruii reviewed (changes requested):
I agree with @JaviGarc1a on his review, I woul also add the following minor change:

nicoloboo02 committed:
fix: Solve bugs to deploy #68

sebasruii committed:
refactor: Add confirmation window #51

JaviGarc1a committed:
fix: Validation error invalid email

JaviGarc1a committed:
refactor: Delete console.log

JaviGarc1a committed:
Merge pull request #69 from ISPP-07/fix/68-Fix_bugs_to_deploy

fix: Solve bugs to deploy #68

JaviGarc1a reviewed (approved):
Great job!

JaviGarc1a committed:
fix: Add validation fields and correct style issue of button

JaviGarc1a committed:
refactor: Abstract Beneficiary details in 2 compontents

BeneficiaryDetails has been divided in two component BeneficiaryDetailsEdit and BeneficiaryDetailsView

JaviGarc1a committed:
Merge branch 'develop' into feature/54-beneficiary_editing_and_deletion

JMGarCas committed:
test: beneficiaryId (not finished)

JMGarCas committed:
Merge branch 'feature/54-beneficiary_editing_and_deletion' of https://github.com/ISPP-07/frontend-acat into feature/54-beneficiary_editing_and_deletion

JaviGarc1a committed:
fix: Change font size input name

JaviGarc1a committed:
fix: Delete @ imports

JMGarCas committed:
test: beneficiaryId (not finished yet)

JMGarCas committed:
Merge branch 'feature/54-beneficiary_editing_and_deletion' of https://github.com/ISPP-07/frontend-acat into feature/54-beneficiary_editing_and_deletion

JMGarCas committed:
test: beneficiaryId

JMGarCas committed:
Merge branch 'develop' into feat/53-Intervention_editing_deletion

JMGarCas committed:
feat: InterventionDetails with update and delete (date not working)

JMGarCas committed:
feat: format date

JMGarCas committed:
fix: delete .yaml to fix deploy

Manuelgruiz committed:
fix:Delete import button  #52

JMGarCas committed:
feat: date now workds

Manuelgruiz committed:
fix:Rewrite export  #57

Ronald752 committed:
feat: Add test

JaviGarc1a committed:
fix: Change format date

Ronald752 committed:
Merge branch 'develop' into feature/51-User_listing_editing_deletion

Ronald752 committed:
Merge pull request #66 from ISPP-07/feature/51-User_listing_editing_deletion

Feature/51 user listing editing deletion

sebasruii committed:
test: Add password update tests #57

nicoloboo02 committed:
test: Add test #52

I have also fixed a
small bug with the export.

nicoloboo02 committed:
Merge pull request #67 from ISPP-07/feature/52-Derecognised_beneficiary

feat:Add rehabilited screen #53

JaviGarc1a committed:
Merge branch 'develop' into feature/54-beneficiary_editing_and_deletion

sebasruii committed:
Merge branch 'develop' into feature/57-Password_recovery

JMGarCas committed:
fix: tests

JMGarCas committed:
Merge pull request #65 from ISPP-07/feature/54-beneficiary_editing_and_deletion

Feature/54 beneficiary editing and deletion

Ronald752 committed:
feat: Add UpdatePassword

Ronald752 committed:
fix: Modify variable name

sebasruii committed:
Merge pull request #72 from ISPP-07/fix/71-Fix_CreateUserForm

fix: Modify variable name

Ronald752 committed:
feat: Add password recovery views #61

sebasruii committed:
Merge pull request #73 from ISPP-07/feature/57-Password_recovery

feat: Add UpdatePassword

JaviGarc1a committed:
Merge branch 'develop' into feat/53-Intervention_editing_deletion

JaviGarc1a committed:
test: Test page intervention details

JaviGarc1a committed:
test: Test InterventionDetailsUpdate

JaviGarc1a committed:
test: Test interventionDetailsView

JaviGarc1a committed:
Merge pull request #70 from ISPP-07/feat/53-Intervention_editing_deletion

Feat/53 intervention editing deletion

JaviGarc1a committed:
fix: Fix tests critical issue codacy

JaviGarc1a committed:
fix: Fix test create user

JaviGarc1a committed:
fix: Fix test intervention card

JaviGarc1a committed:
fix: Add alt to Images

JaviGarc1a committed:
fix: Change exportData function to default export

sebasruii committed:
Merge pull request #75 from ISPP-07/fix/74-bugs_to_deploy

fix: Fix tests critical issue codacy

JaviGarc1a committed:
fix: Fix tests

sebasruii committed:
fix: Fix import #76

sebasruii committed:
Merge pull request #77 from ISPP-07/fix/76-bugs_to_deploy

fix: Fix tests

sebasruii reviewed (approved):
Great work during this sprint!

JaviGarc1a committed:
fix: Create beneficiary and register intervention

TunixR committed:
Merge pull request #82 from ISPP-07/fix/81-bugs_to_deploy

fix: Create beneficiary and register intervention
